dev/doxygen/pages/release_workflow.md

```diff
@@ -0,0 +1,155 @@
+already released versions. So we have to provide such updates in release
+branches. For every major version we create a release branch, which is then used
+for *any* update for that major version. For example on the branch `release/2`
+we would provide the releases "2.0", "2.0.1" (bugfix) and "2.1.0" (new feature).
+
+![Git Branches](release_workflow_diagram.png)
+
+Features and bugfixes which do not affect the file format may be cherry-picked
+from `master` to release branches. This way we can publish updates for the last
+officially released application version while the next major version is still in
+development. Generally we do *not* maintain older releases than the last stable
+release because this would be too much effort. So for example as soon as the
+version "0.2.0" is released, the version "0.1.x" is no longer maintained.
+
+
+# Git Tags
+
+For every officially released version we create a Git tag on the corresponding
+release branch. The tag name contains all three numbers of the application's
+version number. Examples: "0.1.0-rc1", "0.1.0", "1.2.0", "1.2.1"
+
+
+# Warning in Unstable Versions {#doc_release_workflow_warning}
+
+Because (at least) the `master` branch is always unstable, it's very dangerous
+for end users to work productively with this version (it could break projects
+etc.!). So we should warn users in that situation.
+
+This is done by displaying warnings in the application's GUI. Their visibility
+is controlled by the preprocessor define `FILE_FORMAT_STABLE`. It's set to `0`
+(warning visible) on all branches, except on release branches where it's set to
+`1` (warning disabled). As release branches always have a stable file format, it
+should be safe to use *any* state of a release branch (even if not officially
+released).
+
+
+# Translations {#doc_release_workflow_translations}
+
+We use [Transifex] to translate LibrePCB into other languages. As Transifex
+doesn't have support for branches, we use a separate [resource] for each major
+application version. The translations for the `master` branch are contained in
+the resource `librepcb.ts`, while translations for releases are contained in
+resources like `librepcb-0.1.ts`.
+
+On the `master` branch, no translation files are checked into our source
+repository because they change very often (avoid commit spam). But on release
+branches, the translations are checked into the repository so they are always
+available when building official releases (important for packagers). Of course
+these translations are based on the Transifex resource for the corresponding
+major version, *not* on the latest `librepcb.ts` resource.
+
+
+# Changelog {#doc_release_workflow_changelog}
+
+The changelog is not checked into the repository because it would probably lead
+to confusion (add it to the `master` branch, release branches or even both?).
+Instead, we use the [GitHub Releases] page to list the changes of every release
+(resp. tag).
```

Comment:
Maybe it would be good to link those from the website (or even integrate the changelog into the website).

Reply:
Updated documentation accordingly.